### ISSUE TITLE
Fix false negatives for consecutive space for single-line comments in comment-whitespace-inside

### ISSUE THREAD

**Issue by ThiefMaster** [stylelint/stylelint#4776]
> What is the problem you're trying to solve?

My codebase as messy comments like this in some files:

```css
/*
        Default Styles and layout for Conf-pages
        Do not custom this CSS
    */

/*     ------Wrapper------ */

/*     this is the wrapper of whole conference page */

.conf {
  width: 100%;
  border: none;
  margin: 0;
}

/*
    Header
*/
```

> What solution would you like to see?

I would like to have a rule that fixes it like this:

```css
/*
Default Styles and layout for Conf-pages
Do not custom this CSS
*/

/* ------Wrapper------ */

/* this is the wrapper of whole conference page */

.conf {
  width: 100%;
  border: none;
  margin: 0;
}

/*
Header
*/
```

Here are some smaller examples of what should be valid/invalid:

```css
/* valid */

/*
also
valid
*/

/*invalid*/
/*        invalid */

/*
  invalid
  more
  text
*/
```

An edge case is a multiline comment that only consists of a single line. This could be converted to single-line.

I think this behavior could be added to `comment-whitespace-inside` since it is not very useful right now anyway (`never` results in awful style, `always` allows lots of weird consecutive whitespace)

**Comment by jeddy3** [comment]
@ThiefMaster Thanks for the request and for using the template.

Yes, there's a lot we can improve here.

I believe we were liberal with comments because of the many ways people write them. For example, in [`sanitize.css`](https://github.com/csstools/sanitize.css/blob/master/sanitize.css) there are both multiline comments that are spaced and those that are newlined (or sometimes both as shown in the last `*/`):

```
  
/* Document
 * ========================================================================== */

/**
 * Add border box sizing in all browsers (opinionated).
 */
```

So we avoided our usual [`*-newline/space-before/after` approach](https://stylelint.io/user-guide/rules/combine#-newlinespace-before-and--newlinespace-after-rules) that we use for other constructs for something more permissive.

I suggest we address `/*········invalid */` in this rule as a false negative. If a marker is only preceded or succeeded by spaces, i.e. no newlines, then the `"always"` option should enforce a single space. This will make the rule stricter by default. It can, if needed, be made more permissive with an option. We won't add this option until it is requested, though.

We should tackle the indentation within another issue. The existing [`indentation` rule](https://stylelint.io/user-guide/rules/indentation) may make for a suitable blueprint. I imagine we'd call the rule something like `comment-multi-line-indentation` and it'd have an `indentClosingMarker` option, similar to [`indentClosingBrace` one in `indentation`](https://stylelint.io/user-guide/rules/indentation#indentclosingbrace-truefalse).

We're probably opening a can of worms with a rule like this, but it'll be good to have something that is useful for the most common use cases, e.g., doc blocks and flush multiline comments.

I'll open a new issue for this one.

> An edge case is a multiline comment that only consists of a single line. This could be converted to single-line.

This would likely be a rule by itself, with its own edge cases. Feel free to open another issue to discuss, or we can put it on the backburner for now.

**Comment by jeddy3** [comment]
I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
